server/src/db/post_view.rs

```diff
@@ -73,81 +75,35 @@ pub struct PostView {
   pub saved: Option<bool>,
 }
 
-impl PostView {
-  pub fn list(
-    conn: &PgConnection,
-    type_: ListingType,
-    sort: &SortType,
-    for_community_id: Option<i32>,
-    for_creator_id: Option<i32>,
-    search_term: Option<String>,
-    url_search: Option<String>,
-    my_user_id: Option<i32>,
+pub struct PostViewQuery<'a> {
+  conn: &'a PgConnection,
+  query: BoxedQuery<'a, Pg>,
+  my_user_id: Option<i32>,
+  for_creator_id: Option<i32>,
+  page: Option<i64>,
+  limit: Option<i64>,
+}
+
+impl<'a> PostViewQuery<'a> {
+  pub fn create(
+    conn: &'a PgConnection,
+    r#type: ListingType,
+    sort: &'a SortType,
     show_nsfw: bool,
     saved_only: bool,
     unread_only: bool,
-    page: Option<i64>,
-    limit: Option<i64>,
-  ) -> Result<Vec<Self>, Error> {
+  ) -> Self {
```

Comment:
A few requests here:

- Rename this to `PostQueryBuilder`, we'll leave off view since view's are pretty much the main way the data is fetched.
- these 5 params, listing_type, sort, show_nsfw, saved_only, and unread_only, have defaults that are All, Hot, false, false, and false. So these can also be moved up to the builder as `Option`s also. 

It'll look nicer too in the code to have these explicit in the builder pattern. If you need my help to do those as default, let me know.